currencyii/src/main/java/nl/tudelft/trustchain/currencyii/ui/bitcoin/VotesFragment.kt

```diff
@@ -0,0 +1,136 @@
+package nl.tudelft.trustchain.currencyii.ui.bitcoin
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.navigation.fragment.findNavController
+import androidx.viewpager2.widget.ViewPager2
+import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayoutMediator
+import nl.tudelft.ipv8.util.toHex
+import nl.tudelft.trustchain.common.R
+import nl.tudelft.trustchain.common.ui.TabsAdapter
+import nl.tudelft.trustchain.currencyii.ui.BaseFragment
+import nl.tudelft.trustchain.currencyii.sharedWallet.SWTransferFundsAskTransactionData
+
+class VotesFragment : BaseFragment(R.layout.fragment_votes) {
+    private lateinit var tabsAdapter: TabsAdapter
+    private lateinit var viewPager: ViewPager2
+
+    private val TAB_NAMES = arrayOf("Upvotes", "Downvotes", "Undecided votes")
+
+    private lateinit var voters: HashMap<Int, ArrayList<String>>
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_votes, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+
+        val localArgs = arguments
+        if (localArgs is Bundle) {
+            val userID = getTrustChainCommunity().myPeer.publicKey.keyToBin().toHex()
+            val position = localArgs.getInt("position")
+
+            val block = getCoinCommunity().fetchProposalBlocks()[position]
+
+            val rawData = SWTransferFundsAskTransactionData(block.transaction)
+            val data = rawData.getData()
+
+            val walletId = data.SW_UNIQUE_PROPOSAL_ID
+            val priceString = data.SW_TRANSFER_FUNDS_AMOUNT.toString() + " Satoshi"
+
+            val signatures =
+                ArrayList(getCoinCommunity().fetchProposalSignatures(
+                    data.SW_UNIQUE_ID,
+                    data.SW_UNIQUE_PROPOSAL_ID
+                ))
+            voters = rawData.SW_VOTES
+            voters[0] = signatures
+            voters[2]!!.removeAll(signatures)
```

Comment:
if we initialize voters 0,1,2 with list and just append and remove we don't have to use !! everywhere